docs/Mongo-schema.md

```diff
@@ -0,0 +1,151 @@
+# Mongo Schema
+
+Proposed schema:
+
+```
+Scheduler{
+  ref: Tasks {
+    ref: configs {
+    }
+	ref: task_result {
+	}
+    ref: {
+	  schedule_entry
+	}
+  }
+}
+```
+  - A Scheduler object contains **references** to every task.
+  - A task has a **reference** to a config (there exists 3 types of configs).
+  - A task **references** a schedule_entry.
+  - A task **references** to a task result.
+
+## Documents
+
+### Task
+
+```json
+{
+  "_id": "ObjectId()",
+  "task_id": "task_Api-request_63f1bd71-f441-4519-8a41-44643ccb4dad",
+  "creation_time": "04/27/2021 13:35:35",
+  "priority": 0,
+  "type": "Api-request",
+  "configs": ObjectID("config_1"),
+  "task_results": [ObjectID("taskResult_1"), ObjectID("taskResult_2")],
+  "schedule_entry": ObjectID("schedule-entry_1")
+}
```

Comment:
This is just a thought, maybe it would be better to put the `task_id `into the `scheduler_entry` schema instead of the other way around, as the Scheduler class would be the one that looks for the `scheduler_entries` and defines which entries should run.
Because with this current schema you have  it seems that the Scheduler will have to look into all the tasks to check  the `scheduler_entry` information that each task has

Reply:
Thanks, the new schema it is now:

  - A Scheduler object contains **references** to schedule_entry documents.
  - A schedule_entry **references** to a task.  
  - A task has a **reference** to a config (there exists 3 types of configs), and a task_result.